### PR TITLE
refactor: BREAKING adhere to the LedgerPlugin interface

### PR DESCRIPTION
**Pull request by sharafian**
Applies changes that make the plugin pass tests in [Ilp-Plugin-Tests](https://github.com/interledger/js-ilp-plugin-tests). The only tests that fail are sending a transfer with an amount of 0, and the tests for the `replyToTransfer` function, which has not been implemented.

### CODE REVIEW

File: src/lib/plugin.js [hunk 1]
```diff
@@ -308,17 +319,23 @@ class FiveBellsLedger extends EventEmitter2 {
         })
 
         if (res.statusCode !== 200) {
+          this.emit('reject', transfer,
+            new Buffer('Unexpected status code: ' + res.statusCode))
           throw new Error('Unexpected status code: ' + res.statusCode)
```
Comment (sentientwaffle) [inline src/lib/plugin.js:347]: Is there a reason to throw an error _and_ emit `"reject"`?

Comment (sharafian) [inline src/lib/plugin.js:347]: I'll remove the error throwing, but the `reject` event is required for the LPI

Comment (sentientwaffle) [inline src/lib/plugin.js:347]: Then you need to wrap the `request()` in a try/catch as well -- connection errors will still throw.

File: src/lib/plugin.js [hunk 2]
```diff
@@ -212,7 +213,9 @@ class FiveBellsLedger extends EventEmitter2 {
 
     return {
       precision: res.body.precision,
-      scale: res.body.scale
+      scale: res.body.scale,
+      currencyCode: res.body.currencyCode || null,
```
Comment (justmoon) [inline src/lib/plugin.js]: Should be `res.body.currency_code`.

Comment (sharafian) [inline src/lib/plugin.js]: fixing

**Comment by justmoon** [comment]
Should be updated with the comments from here: https://github.com/interledger/rfcs/issues/45#issuecomment-234931360

I.e. `send()` should throw if it wasn't able to prepare the transfer successfully.

File: test/indexSpec.js [hunk 3]
```diff
@@ -599,17 +714,14 @@ describe('PluginBells', function () {
...
-        } catch (err) {
-          assert(err instanceof ExternalError)
-          assert.equal(err.message, 'Remote error: status=400')
-        }
+
+        this.plugin.send({
+          id: '6851929f-5a91-4d02-b9f4-4ae6b7f1768c',
+          account: 'red.alice',
+          amount: '123'
+        }).catch(() => {
```
Comment (emschwartz) [inline test/indexSpec.js]: You're not actually checking what type of error is being thrown here

File: test/indexSpec.js [hunk 4]
```diff
@@ -640,6 +752,22 @@ describe('PluginBells', function () {
...
+          .post('/targets', ['http://red.example/transfers/6851929f-5a91-4d02-b9f4-4ae6b7f1768c/fulfillment'])
+          .reply(400)
+
+        this.plugin.send({
+          id: '6851929f-5a91-4d02-b9f4-4ae6b7f1768c',
+          account: 'red.alice',
+          amount: '123',
+          cases: ['http://notary.example/cases/2cd5bcdb-46c9-4243-ac3f-79046a87a086']
+        }).catch((err) => {
+          console.error(err)
```
Comment (emschwartz) [inline test/indexSpec.js]: You shouldn't have log statements here and you should also check what the error is

**Comment by emschwartz** [comment]
LGTM

**Pull request merged**